### DISCUSSION TITLE
Performance metrics of multiple API requests reported together. How to get performance metrics for each API request ?

### DISCUSSION THREAD

**Discussion by rrnareshkumar** [artilleryio/artillery#2120] [Artillery]
Hi , I am doing load testing for 2 API requests (please refer the config and scenario details below) . I generated json and html reports  post test execution completion and noticed that performance metrics such response times , https requests count , responses count , http.codes.200 count , bytes download count are reported together . I would like to segregate and view performance metrics separately for each API request (in this case for addtocart and place order) . 

Please suggest what changes needs to be done to get  performance metrics separately for each API request

config:
  target: https://testsite.com/
  phases:
    - duration: 60
      arrivalRate: 1
    - duration: 30
      arrivalRate: 1
scenarios:
  - flow:
      - get:
          url: /addtocart
      - get:
          url: /placeorder

**Comment by InesNi** [comment]
Hi @rrnareshkumar 👋🏻 !

Have you tried enabling the built-in [metrics-by-endpoint plugin](https://www.artillery.io/docs/reference/extensions/metrics-by-endpoint) like I suggested in the other discussion? It should be what you need for this 😃

Hope it helps!
